### PR TITLE
Use obs box in ScalarWave, switch ObserveVolumeIntegrals to obs box

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Add ObservationBox compute tag support to `ObserveVolumeIntegrals`
- Switch the scalar wave executable to using the observation box instead of storing constraints, L2 norms, etc. in the DataBox.

Depends on:
- [x] #3558 

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp
Comment (nilsvu) [inline src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp]: factor the `AnalyticCompute` into a type alias as well, so this formats better?

**Review by nilsvu — DISMISSED** [review]
LGTM, no need for my re-approval if you do the one suggestion and rebase on #3558

**Comment by nilsdeppe** [comment]
Rebased and added the type alias. Thanks for the review :)

**Review by nilsvu — APPROVED** [review]

**Pull request merged**
